Update connect-your-database.md (#2437)
fishtown-analytics/docs.getdbt.com: website/docs/docs/get-started/connect-your-database.md

### Connecting via an SSH Tunnel

To connect to a Postgres, Redshift, or AlloyDB instance via an SSH tunnel, select the **Use SSH Tunnel** option when creating your connection. When configuring the tunnel, you must supply the hostname, username, and port for the [bastion server](#about-the-bastion-server-in-aws).

Once the connection is saved, a public key will be generated and displayed for the Connection. You can copy this public key to the bastion server to authorize dbt Cloud to connect to your database via the bastion server.

<Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/postgres-redshift-ssh-tunnel.png" title="A generated public key for a Redshift connection"/>

#### About the Bastion server in AWS

<details>
  <summary>What is a Bastion server?</summary>
  <div>
    <div>A bastion server in <a href="https://aws.amazon.com/blogs/security/how-to-record-ssh-sessions-established-through-a-bastion-host/">Amazon Web Services (AWS)</a> is a host that allows dbt Cloud to open an SSH connection. <br></br>
    

> dbt Cloud only sends queries and doesn't transmit large data volumes. This means the bastion server can run on an AWS instance of any size, like a t2.small instance or t2.micro.<br></br>

Reviewer: ```suggestion
dbt Cloud only sends queries and doesn't transmit large data volumes. This means the bastion server can run on an AWS instance of any size, like a t2.small instance or t2.micro.<br></br><br></br>
```

    
Make sure the location of the instance is the same Virtual Private Cloud (VPC) as the Redshift instance, and configure the security group for the bastion server to ensure that it's able to connect to the warehouse port.
    </div>
  </div>
</details>

#### Configuring the Bastion Server in AWS:

To configure the SSH tunnel in dbt Cloud, you'll need to provide the hostname/IP of your bastion server, username, and port, of your choosing, that dbt Cloud will connect to. Review the following steps:

- Verify the bastion server has its network security rules set up to accept connections from the [dbt Cloud IP addresses](/docs/deploy/regions-ip-addresses) on whatever port you configured.
- Set up the user account by using the bastion servers instance's CLI, The following example uses the username `dbtcloud:`
    

>     `sudo groupadd dbtcloud`

Reviewer: ```suggestion
    `sudo groupadd dbtcloud`<br></br>
```

    
    `sudo useradd -m -g dbtcloud dbtcloud`
    
    `sudo su - dbtcloud`
    
    `mkdir ~/.ssh`
    
    `chmod 700 ~/.ssh`
    
    `touch ~/.ssh/authorized_keys`
    
    `chmod 600 ~/.ssh/authorized_keys`
    
- Copy and paste the dbt Cloud generated public key, into the authorized_keys file.

> With the Bastion EC2 instance in place, and the user properly configured, you should be able to successfully connect to the Redshift environment.

Reviewer: ```suggestion
The Bastion server should now be ready for dbt Cloud to use as a tunnel into the Redshift environment.
```

<Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/postgres-redshift-ssh-tunnel.png" title="A generated public key for a Redshift connection"/>

> #### About the Bastion server in AWS:

Reviewer: ```suggestion

```

> #### About the Bastion server in AWS:

Reviewer: ```suggestion
#### About the Bastion server in AWS
```

> A bastion server is really just a host that dbt Cloud can open up an SSH connection to, and given that dbt Cloud is generally only sending queries, and not transmitting large data volumes, you can safely assume that the bastion server can run on a small AWS instance. I'd offer the following recommendations:

Reviewer: suggesting making this into a toggle and summarizing. also is my sug below technically accurate?

>dbt Cloud only sends queries and doesn't transmit large data volumes, which means the bastion server can run on an AWS instance of any size

```suggestion
<details>
  <summary>What is a Bastion server?</summary>
  <div>
    <div>A bastion server in <a href="https://aws.amazon.com/blogs/security/how-to-record-ssh-sessions-established-through-a-bastion-host/">Amazon Web Services (AWS)</a> is a host that allows dbt Cloud to open an SSH connection. <br></br>
    
dbt Cloud only sends queries and doesn't transmit large data volumes. This means the bastion server can run on an AWS instance of any size, like a t2.small instance or t2.micro.<br></br>
    
Make sure the location of the instance is the same Virtual Private Cloud (VPC) as the Redshift instance, and configure the security group for the bastion server to ensure that it's able to connect to the warehouse port.
    </div>
  </div>
</details>
```

> - A t2.small instance is likely sufficiently sized for this purpose (even a t2.micro will probably suffice, in all honesty)

Reviewer: ```suggestion

```

> - Make sure the instance is located in the same VPC as the Redshift instance, and configure the security group for the bastion server to ensure that it is able to connect to the warehouse port

Reviewer: ```suggestion

```

#### Configuring the Bastion Server in AWS:

> When you configure the tunnel in dbt Cloud, you'll need to provide the hostname/IP of your bastion server, as well as a username and port, of your choosing, that dbt Cloud will connect to:

Reviewer: ```suggestion
To configure the SSH tunnel in dbt Cloud, you'll need to provide the hostname/IP of your bastion server, username, and port, of your choosing, that dbt Cloud will connect to. Review the following steps:
```

> - Make sure that the bastion server has its network security rules set up properly to accept connections from the dbt Cloud IP addresses on whatever port you configure for the bastion server in dbt Cloud

Reviewer: ```suggestion
- Verify the bastion server has its network security rules set up to accept connections from the [dbt Cloud IP addresses](/docs/deploy/regions-ip-addresses) on whatever port you configured.
```

> - You'll need to set up the user account (I'm using the username dbtcloud below) via the bastion servers instance's CLI:

Reviewer: ```suggestion
- Set up the user account by using the bastion servers instance's CLI, The following example uses the username `dbtcloud:`
```

>     `sudo groupadd dbtcloud`

Reviewer: hey @will-sargent-dbtlabs is the example suppose to look like this screenshot?(2 lines) or is it supposed to take up 6 lines?
![Screenshot 2022-12-13 at 14 58 52](https://user-images.githubusercontent.com/89008547/207367974-2d18d2dd-a89e-403b-b427-c78b1d579213.png)

Author: 6 line is better @mirnawong1 , thanks!

    
    `sudo useradd -m -g dbtcloud dbtcloud`
    
    `sudo su - dbtcloud`
    
    `mkdir ~/.ssh`
    
    `chmod 700 ~/.ssh`
    
    `touch ~/.ssh/authorized_keys`
    
    `chmod 600 ~/.ssh/authorized_keys`
    

> - You'll need to copy-paste the generated public key from dbt Cloud into the authorized_keys file

Reviewer: ```suggestion
- Copy and paste the dbt Cloud generated public key, into the authorized_keys file.
```

> With the Bastion EC2 instance in place, and the user properly configured, you should be able to successfully connect to the Redshift environment.

Reviewer: hey @will-sargent-dbtlabs - should the Bastion EC2 instance be referenced earlier in the doc? I'm trying to understand why we're specifying EC2 instance at this stage?

Author: Yeah, this summary line isn't quite there..

Author: @mirnawong1 maybe something like:  "The Bastion server should now be ready for dbt Cloud to use as a tunnel into the Redshift environment."
